integration-libs/cdc/root/service/cdc-js.service.ts

```diff
@@ -170,81 +170,112 @@ export class CdcJsService implements OnDestroy {
-  onInitRegistrationHandler(user: UserSignUp, response: any) {
-    if (response && response.regToken && user.uid && user.password) {
-      (this.winRef.nativeWindow as { [key: string]: any })?.[
-        'gigya'
-      ]?.accounts?.register({
-        email: user.uid,
-        password: user.password,
-        profile: {
-          firstName: user.firstName,
-          lastName: user.lastName,
-        },
-        regToken: response.regToken,
-        finalizeRegistration: true,
-        ignoreInterruptions: true,
-        callback: (response: any) => {
-          this.zone.run(() => {
-            if (response && response.status === 'FAIL') {
-              this.handleRegisterError(response);
-            }
-          });
-        },
-      });
-    }
+  protected onInitRegistrationHandler(
+    user: UserSignUp,
+    response: any
+  ): Observable<Boolean> {
+    return new Observable<Boolean>((isRegistered) => {
+      if (response && response.regToken && user.uid && user.password) {
+        (this.winRef.nativeWindow as { [key: string]: any })?.[
+          'gigya'
+        ]?.accounts?.register({
+          email: user.uid,
+          password: user.password,
+          profile: {
+            firstName: user.firstName,
+            lastName: user.lastName,
+          },
+          regToken: response.regToken,
+          finalizeRegistration: true,
+          callback: (response: any) => {
+            this.zone.run(() => {
+              if (response && response.status === 'FAIL') {
+                this.handleRegisterError(response);
+                isRegistered.next(false);
+              } else if (response && response.status === 'OK') {
+                isRegistered.next(true);
+              }
+              isRegistered.complete();
+            });
+          },
+        });
+      }
+    });
   }
 
   /**
    * Trigger CDC User log in using CDC APIs.
    *
    * @param response
    */
-  loginUserWithoutScreenSet(email: string, password: string, response: any) {
-    if (response) {
-      (this.winRef.nativeWindow as { [key: string]: any })?.[
-        'gigya'
-      ]?.accounts?.login({
-        loginID: email,
-        password: password,
-        callback: (response: any) => {
-          this.zone.run(() => this.handleLoginError(response));
-        },
-      });
-    }
+  loginUserWithoutScreenSet(
+    email: string,
+    password: string,
+    response: any
+  ): Observable<Boolean> {
+    return new Observable<Boolean>((isLoggedIn) => {
+      if (response) {
+        (this.winRef.nativeWindow as { [key: string]: any })?.[
+          'gigya'
+        ]?.accounts?.login({
+          loginID: email,
+          password: password,
+          callback: (response: any) => {
+            this.zone.run(() => {
+              if (response && response.status === 'FAIL') {
+                this.handleLoginError(response);
+                isLoggedIn.next(false);
+              } else if (response && response.status === 'OK') {
```

Comment:
Just a question: can the response `status` be different than 'FAIL' or 'OK'? If yes, then our method will just complete the stream without emitting any value or error.

Same question applies also for: Register and Forgot Password